scrapy_po/middleware.py

```diff
@@ -21,45 +23,44 @@ class InjectPageObjectsMiddleware:
     """
     @inlineCallbacks
     def process_response(self, request: Request, response, spider):
-        # find out which arguments need to be created
+        # find out the dependencies
         callback = get_callback(request, spider)
-        kwargs_to_provide = andi.to_provide(callback,
-                                            can_provide=self.is_page_object)
+        providers = build_providers(response)
+        can_provide = can_provide_fn(providers)
```

Comment:
I think it'd be a bit more clear what's going on if you pass providers.keys() instead of just providers

Reply:
Done here: https://github.com/scrapinghub/scrapy-po/pull/5/commits/166f48fe3b47fc7963629867bff9010b382c1bef